src/sentry/integrations/utils/metrics.py

```diff
@@ -423,3 +424,43 @@ def get_integration_name(self) -> str:
 
     def get_interaction_type(self) -> str:
         return str(self.interaction_type)
+
+
+@contextlib.contextmanager
+def with_event_lifecycle(
+    metric: EventLifecycleMetric,
+    assume_success: bool = True,
+    sample_log_rate: float = 1.0,
+    halt_on_exceptions: list[type[BaseException]] | None = None,
+):
+    """
+    Decorator that wraps a function with event lifecycle tracking.
+    This decorator provides the same functionality as using the EventLifecycle
+    context manager manually, but in a more convenient decorator form.
+    Args:
+        metric: The EventLifecycleMetric instance to use for tracking
+        assume_success: Whether to assume success if no explicit outcome is recorded (default True)
+        sample_log_rate: Rate at which to sample logs (0.0-1.0)
+        halt_on_exceptions: List of exception types that should be treated as halts instead of failures
+    Example:
+        @with_event_lifecycle(
+            my_metric,
```

Comment:
Could we add some more examples on what `my_metric` would be? Like for GitHub would it be 

```
       @with_event_lifecycle(
            IntegrationPipelineViewEvent(
        interaction_type=IntegrationPipelineViewType.OAUTH_LOGIN, 
        domain=IntegrationDomain.SOURCE_CODE_MANAGEMENT,
        provider_key=GitHubIntegrationProvider.key),
             halt_on_exceptions=[...],
```